mlir/lib/Dialect/Arith/IR/ValueBoundsOpInterfaceImpl.cpp

```diff
@@ -75,6 +75,19 @@ struct MulIOpInterface
   }
 };
 
+struct DivSIOpInterface
+    : public ValueBoundsOpInterface::ExternalModel<DivSIOpInterface, DivSIOp> {
+  void populateBoundsForIndexValue(Operation *op, Value value,
+                                   ValueBoundsConstraintSet &cstr) const {
+    auto divSIOp = cast<DivSIOp>(op);
+    assert(value == divSIOp.getResult() && "invalid value");
+
+    AffineExpr lhs = cstr.getExpr(divSIOp.getLhs());
+    AffineExpr rhs = cstr.getExpr(divSIOp.getRhs());
+    cstr.bound(value) == lhs.floorDiv(rhs);
```

Comment:
... `divsi` isn't `floordiv`?

That is,I think this'll be wrong for negative numbers.

Reply:
Oh right, round towards zero... and this doesn't work for divui because affine floor/ceildiv is signed. I can't think of any way to support this then.

Reply:
I'm pretty sure there's a way to build round to 0 from round to minus infinity, but I'm not sure about the math there

Reply:
Switched to floordivsi, which should be the same semantics.